From 33f44cb0b715a9bac435754c635b0d49e0ab4c34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Antti=20M=C3=A4=C3=A4tt=C3=A4?= <antti.maatta@qt.io>
Date: Tue, 23 Sep 2025 14:16:18 +0300
Subject: [PATCH 267/553] tracepointgen: Add file path to include directories

Add the parsed file absolute path to include directories so that
files in the same directory can be found when resolving includes.

Fixes: QTBUG-140443
Change-Id: I91f913173dc42fd6b954a1821d992ad6288f694b
Reviewed-by: Samuli Piippo <samuli.piippo@qt.io>
(cherry picked from commit 5123ac724a70b1a100a0cc31d40c16cb93ba90b7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/tools/tracepointgen/tracepointgen.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/tools/tracepointgen/tracepointgen.cpp b/src/tools/tracepointgen/tracepointgen.cpp
index d814c69873e..967a9f78159 100644
--- a/src/tools/tracepointgen/tracepointgen.cpp
+++ b/src/tools/tracepointgen/tracepointgen.cpp
@@ -4,6 +4,7 @@
 #include <qstring.h>
 #include <qstringlist.h>
 #include <qfile.h>
+#include <qfileinfo.h>
 
 #include "tracepointgen.h"
 #include "parser.h"
@@ -50,6 +51,8 @@ int main(int argc, char *argv[])
                     qPrintable(inputFile), qPrintable(in.errorString()));
         }
         DEBUGPRINTF(printf("tracepointgen: parse %s\n", qPrintable(inputFile)));
+        QFileInfo info(inputFile);
+        parser.addIncludeDirs({info.absolutePath()});
         parser.parse(in, inputFile);
     }
     if (parser.isEmpty())
-- 
2.51.2

